### PR TITLE
Adding JMX functionalities to the AS400 PCML Connector 

### PR DESCRIPTION
**Pull request by markelari**
## Purpose
Adding JMX functionalities to the AS400 PCML Connector
Fixes: https://github.com/wso2/api-manager/issues/1833

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: src/main/java/org/wso2/carbon/connector/pcml/PCMLPool.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (SanojPunchihewa) [inline src/main/java/org/wso2/carbon/connector/pcml/PCMLPool.java:2]:
```suggestion
 * Copyright (c) 2023, WSO2 LLC. (http://www.wso2.org) All Rights Reserved.
```

File: src/main/java/org/wso2/carbon/connector/pcml/PCMLPool.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
```
Comment (SanojPunchihewa) [inline src/main/java/org/wso2/carbon/connector/pcml/PCMLPool.java:4]:
```suggestion
 *  WSO2 LLC. licenses this file to you under the Apache License,
```

File: src/main/java/org/wso2/carbon/connector/pcml/PCMLPool.java (hunk 1 above)
Comment (SanojPunchihewa) [reply src/main/java/org/wso2/carbon/connector/pcml/PCMLPool.java:2]: Update the PCMLPoolMBean class as well

File: src/main/java/org/wso2/carbon/connector/pcml/PCMLPool.java [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+        this.as400ConnectionPool = connectionPool;
+        this.as400 = as400;
+    }
+
+    @Override
+    public int getMaxConnections() {
+        return this.getAs400ConnectionPool().getMaxConnections();
+    }
+
+    public AS400ConnectionPool getAs400ConnectionPool() {
```
Comment (SanojPunchihewa) [inline src/main/java/org/wso2/carbon/connector/pcml/PCMLPool.java:49]: Do we need this getter?

**Comment by SanojPunchihewa** [comment]
Shall we link the git issue in the description and update the PR title with a more meaningful title

File: src/main/java/org/wso2/carbon/connector/pcml/PCMLPool.java (hunk 1 above)
Comment (markelari) [reply src/main/java/org/wso2/carbon/connector/pcml/PCMLPool.java]: done

File: src/main/java/org/wso2/carbon/connector/pcml/PCMLPool.java (hunk 2 above)
Comment (markelari) [reply src/main/java/org/wso2/carbon/connector/pcml/PCMLPool.java]: done

File: src/main/java/org/wso2/carbon/connector/pcml/PCMLPool.java (hunk 3 above)
Comment (markelari) [reply src/main/java/org/wso2/carbon/connector/pcml/PCMLPool.java]: checked and its not being called, deleted it

File: src/main/java/org/wso2/carbon/connector/pcml/PCMLPool.java [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+    public boolean isPretestConnections() {
+        return this.as400ConnectionPool.isPretestConnections();
+    }
+
+    @Override
+    public int getActiveConnectionCount() {
+        return this.as400ConnectionPool.getActiveConnectionCount(as400.getSystemName(), as400.getUserId());
+    }
+}
```
Comment (SanojPunchihewa) [inline src/main/java/org/wso2/carbon/connector/pcml/PCMLPool.java:85]: Shall we add a new line here?

Comment (markelari) [reply src/main/java/org/wso2/carbon/connector/pcml/PCMLPool.java:85]: done

**Review by SanojPunchihewa — APPROVED** [review]

**Pull request merged**